decidim/decidim
# Add attachments to debate

#### :tophat: What? Why?
This pull request introduces the ability to attach files when creating or editing debates. The proposed feature enables file attachments for debates, both from the admin panel and the frontend, improving the flexibility and functionality of debates within the platform.

#### :pushpin: Related Issues
- Fixes #13479

#### Testing
1. As an admin, go to the admin panel and configure a Debate component.
   - Verify that there is a new setting "Allow attachments" in the configuration.
   - Enable this setting and try to create or edit a debate, ensuring that attachments can be uploaded.
   - Verify that uploaded attachments appear as downloadable links in the debate view.

2. As a participant, try to create or edit a debate from the frontend.
   - Verify that, if the admin has enabled "Allow attachments", you can upload attachments during debate creation or editing.

3. Test file type and size validation by uploading files that either exceed the configured limits or are of unsupported types, ensuring the system shows appropriate error messages.

### :camera: Screenshots
<img width="600" alt="2024-11-04 в 11 42 09" src="https://github.com/user-attachments/assets/beefd8c1-f730-473f-8995-09b1cdd7b356">
<img width="600" alt="2024-11-04 в 11 42 31" src="https://github.com/user-attachments/assets/54226934-2df8-47b0-b75d-0259a376b379">


:hearts: Thank you!
